test/core-test/quantum_objects_evo.jl

```diff
@@ -165,32 +165,51 @@
         a = destroy(N)
         coef1(p, t) = exp(-1im * p.ω1 * t)
         coef2(p, t) = sin(p.ω2 * t)
+        coef3(p, t) = sin(p.ω3 * t)
         t = rand()
-        p = (ω1 = rand(), ω2 = rand())
+        p = (ω1 = rand(), ω2 = rand(), ω3 = rand())
 
         # Operator
         H_td = QobjEvo(((a, coef1), a' * a, (a', coef2)))
         H_ti = coef1(p, t) * a + a' * a + coef2(p, t) * a'
         ψ = rand_ket(N)
         @test H_td(p, t) ≈ H_ti
+        @test iscached(H_td) == true
+        H_td = cache_operator(H_td, ψ)
+        @test iscached(H_td) == true
         @test H_td(ψ, p, t) ≈ H_ti * ψ
         @test isconstant(a) == true
         @test isconstant(H_td) == false
         @test isconstant(QobjEvo(a)) == true
         @test isoper(H_td) == true
 
         # SuperOperator
-        c_ops = [sqrt(rand()) * a]
-        L_td = liouvillian(H_td, c_ops)
-        L_td2 = -1im * spre(H_td) + 1im * spost(H_td) + lindblad_dissipator(c_ops[1])
+        X = a * a'
+        c_op1 = QobjEvo(((a', coef1),))
+        c_op2 = QobjEvo(((a, coef2), (X, coef3)))
+        c_ops = [c_op1, c_op2]
```

Comment:
The two `QobjEvo` have different types. Using `Vector` would introduce some type instabilities. Although it is still safe here in the test, I think that we should always use type stable codes. So far we safely used `Vector` because there was no need. But here we should require to use `Tuple` instead. Speaking of, do we have a check for type inference of the `liouvillian` and `lindblad_dissipator` functions?

Reply:
I can do it tomorrow.